pkg/api/controller.go

```diff
@@ -2716,7 +2716,7 @@ func (c *Controller) MergeIntoBranch(w http.ResponseWriter, r *http.Request, bod
 		writeError(w, http.StatusPreconditionFailed, err)
 		return
 	case errors.Is(err, catalog.ErrConflictFound) || errors.Is(err, graveler.ErrConflictFound):
-		writeError(w, http.StatusConflict, newMergeResultFromCatalog(res))
```

Comment:
Consider checking if we can change the writeError signature to work only on errors

Reply:
Checked this option - but there are many cases that we serialize the different messages and not errors when we send a status code.